storage/database/db_manager.go

```diff
@@ -476,12 +476,62 @@ func (dbm *databaseManager) NewBatch(dbEntryType DBEntryType) Batch {
 		defer dbm.lockInMigration.RUnlock()
 
 		if dbm.inMigration {
-			return dbm.GetStateTrieMigrationDB().NewBatch()
+			newDBBatch := dbm.getDatabase(StateTrieMigrationDB).NewBatch()
+			oldDBBatch := dbm.getDatabase(StateTrieDB).NewBatch()
+			return NewStateTrieDBBatch([]Batch{oldDBBatch, newDBBatch})
 		}
+	} else if dbEntryType == StateTrieMigrationDB {
+		return dbm.GetStateTrieMigrationDB().NewBatch()
 	}
 	return dbm.getDatabase(dbEntryType).NewBatch()
 }
 
+func NewStateTrieDBBatch(batches []Batch) Batch {
+	return &stateTrieDBBatch{batches: batches}
+}
+
+type stateTrieDBBatch struct {
+	batches []Batch
+}
+
+func (stdBatch *stateTrieDBBatch) Put(key []byte, value []byte) error {
+	var errResult error
+	for _, batch := range stdBatch.batches {
+		if err := batch.Put(key, value); err != nil {
+			errResult = err
+		}
+	}
+
+	return errResult
+}
+
+// ValueSize is called to determine whether to write batches when it exceeds
+// certain limit. partitionedDB returns the largest size of its batches to
+// write all batches at once when one of batch exceeds the limit.
+func (stdBatch *stateTrieDBBatch) ValueSize() int {
+	maxSize := 0
+	for _, batch := range stdBatch.batches {
+		if batch.ValueSize() > maxSize {
+			maxSize = batch.ValueSize()
+		}
+	}
+
+	return maxSize
+}
+
+// Write passes the list of batch tasks to taskCh so batch can be processed
+// by underlying workers. Write waits until all workers return the result.
```

Comment:
The above comment seems unrelated to this function. Maybe copied from partitioned database batch?

Reply:
Right. I modified. :) Thanks.